propagators/trace_context_test.go

```diff
@@ -227,15 +223,15 @@ func TestInjectTraceContextToHTTPReq(t *testing.T) {
 				SpanID:     spanID,
 				TraceFlags: otel.FlagsSampled,
 			},
-			wantHeader: "00-4bf92f3577b34da6a3ce929d0e0e4736-0000000000000001-01",
+			wantHeader: "00-4bf92f3577b34da6a3ce929d0e0e4736-0000000000000002-01",
```

Comment:
I think this is incremented because the default Tracer is being reused by different tests. That might not be ideal. If other tests change this test will be impacted.

It might be better to update the `DefaultTracer` function to take a tracer name as a string to distinguish the returned tracer to avoid this time of confusion.

Reply:
As far as I can tell, the span IDs in the header should be incremented (within `TestInjectTraceContextToHTTPReq`), as this was the old behavior as well.

My understanding is that the difference is caused by the fact that the`MockTracer` was [starting span ID from 0](https://github.com/open-telemetry/opentelemetry-go/blob/e22a48de5a1353fb5450783ca9bb5edf651e9849/oteltest/mock_tracer.go#L35), whereas `Tracer` [starts from 1](https://github.com/open-telemetry/opentelemetry-go/blob/e22a48de5a1353fb5450783ca9bb5edf651e9849/oteltest/config.go#L28). This necessitated incrementing IDs in the expected header in test cases below by 1.

Reply:
Ah, good point. I missed that. SGTM